### ISSUE TITLE
"Relationship could not be deleted!"

### ISSUE THREAD

**Issue by Californian** [hasura/graphql-engine#9713]
### Version Information

Server Version: `v2.27.0-cloud.1`

### Environment

Cloud
Env vars: `HASURA_GRAPHQL_EXPERIMENTAL_FEATURES` = `naming_convention`, graphql format is turned on.

### What is the current behaviour?

Dropping an array relationship fails with the following message:
```
Relationship could not be deleted!
Error when parsing command drop_relationship.
See our documentation at https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/index.html#metadata-apis.
Internal error message: key "table" not found
```

### What is the expected behaviour?

Dropping relationships succeed.

### How to reproduce the issue?

1. Create a project.
2. Create three tables, two with uuid ID primary keys (tables "A" and "B"), one with a compound primary key that references the other two (table "C").
3. Add an array relationship on "A" to "C".
4. Attempt to drop the above relationship.

### Screenshots or Screencast
![Screenshot from 2023-06-15 13-34-10](https://github.com/hasura/graphql-engine/assets/1079539/c03d9a2a-43cd-4a7a-b63c-c9ad258bf3ee)

### Please provide any traces or logs that could help here.

400 response from /metadata:

```
{
    "code": "parse-failed",
    "error": "Error when parsing command drop_relationship.\nSee our documentation at https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/index.html#metadata-apis.\nInternal error message: key \"table\" not found",
    "path": "$.args"
}
```

### Any possible solutions/workarounds you're aware of?

No.

### Keywords

Drop relationship, Internal error message: key "table" not found

**Comment by Californian** [comment]
@vijayprasanna13 I can't confirm whether it also happens on the former at the moment because all of my projects are down (separate issue), but this certainly happens from the `Data > [Table] > Relationships` screen.

I'll also note that object relationships all describe the foreign key relationship properly, but the array relationships have nothing in that column. I don't recall whether that is atypical though.

**Comment by Californian** [comment]
Also, I attempted to switch back to `hasura-default` and that did not resolve the issue. I can try starting a new project and not adding that experimental variable.

I'm using GCP and Neon, if that's helpful.

**Comment by vijayprasanna13** [comment]
Thanks @Californian for reverting back quickly on this. 

> Also, I attempted to switch back to hasura-default naming convention and that did not resolve the issue. I can try starting a new project and not adding that experimental variable.

I don't the GQL configuration affects this any way, We've tracked this issue down to the part where the console merges metadata with the database introspection result. This seems to be a front-end issue on our side and we've put this on priority. We're working this out and I'll keep you posted here. 

If you are blocked by this issue on your side, you can try toggling this feature flag off and use the older version of the relationships UI if that solves the issue for you
![Screenshot 2023-06-22 at 12 44 48 PM](https://github.com/hasura/graphql-engine/assets/11921040/a044a5f5-f4da-4c4a-a92e-651838e4915e)

**Comment by vijayprasanna13** [comment]
@Californian we've fixed this in https://github.com/hasura/graphql-engine/commit/9179135148da4646d9f6377d1760d7cf64655b61 and this should be out in `v2.29`
